### ISSUE TITLE
Include API GW path parameters as event parameters.

### ISSUE THREAD

**Issue by jthomas** [apache/incubator-openwhisk#3593]
This is a feature enhancement for the recently introduced path parameters feature in the API GW: https://github.com/apache/incubator-openwhisk/pull/2966

When actions are invoked with path parameters, the full path is included in the event parameters using the `__ow_path` value. **Individual parameterised path values are not automatically parsed from the request and include as separate event parameters.**

It would be a better developer experience if those path parameters were passed as separate values, saving the developer having to manually re-parse them from the `__ow_path` string.

One option would be to introduce a new variable `__ow_path_params` which contained these values.

AWS Lambda supports this [behaviour](https://stackoverflow.com/questions/31329958/how-to-pass-a-querystring-or-route-parameter-to-aws-lambda-from-amazon-api-gatew).

User migrating from that platform posted the following in Slack.

_I'm migrating from AWS Lambda, and struggling somewhat with passing path parameters into a Python handler from the API gateway. In Lambda, I simply do `pathID = event['pathParameters']['id']` but I can't find any documentation on an easy way to do this in Cloud Functions (noting that the dictionary passed into the handler only seems to contain string parameters). Do I need to use regex on `__ow_path` or am I missing something?_

**Comment by rabbah** [comment]
does API gw handle parsing these already?

**Comment by csantanapr** [comment]
This issue should be move to apigateway bulk of the work is on the gateway to parse the path and send the fields in a header to the controller

**Comment by ddragosd** [comment]
Just the confirm an assumption: the user should already map somewhere the path params, correct ? I.e in the “paths” section in swagger.

**Comment by ddragosd** [comment]
@csantanapr I think that if the Gateway reads the path params to pass them via querystrings to the controller, then controller’s impl doesn’t have to be updated to read from the request headers instead; the controller would decorate the incoming payload with the query string values automatically.

**Comment by csantanapr** [comment]
@ddragosd 
Yes they are already in mapped today in swagger config for the operation

The problems I see for using query is conflicting with actual query parameters. 
?name=Carlos
The user doesn’t know if name is referring to a query, or path param 
In cases where /api/:name/profile?name=

If we really want to use query we need to namespace them with something like _ow
Then gateway would send to the web action /api/Alberto/profile?name=Carlos&_ow_path_name=Alberto
With this then no controller changes and user get all path parameters from query parameters that start with some prefix _ow_path_*

Is this what your were proposing?

Again like I said an issue to be discuss on what’s best for apigateway to implement t 
Send path parameters via a header or custom query parameters

**Comment by mdeuser** [comment]
another proposal from another issues/discussion
> a new header __ow_path_parameters that includes a json stringified object having fields names corresponding to the actual path parameter ids and field values corresponding to the path parameter values in the actual url
